### PR TITLE
Bug 1215877 – Preference to enable/disable saving of logins.

### PR DESCRIPTION
**Pull request by jhugman**
We aren't able to easily disable finding logins on the page – because WKUserScripts cannot be removed easily from WKUserController.

### CODE REVIEW

File: Client/Frontend/Browser/LoginsHelper.swift [hunk 1]
```diff
@@ -59,7 +59,7 @@ class LoginsHelper: BrowserHelper {
                    let requestId = res["requestId"] as? String {
                     requestLogins(login, requestId: requestId)
                 }
-            } else if type == "submit" {
+            } else if type == "submit" && self.profile.prefs.boolForKey("saveLogins") ?? true {
```
Comment (fluffyemily) [inline Client/Frontend/Browser/LoginsHelper.swift]: nit: I prefer to have the second conditional in brackets. Makes it easier to read and much clearer.

`&& (self.profile.prefs.boolForKey("saveLogins") ?? true)`

File: Client/Frontend/Settings/SettingsTableViewController.swift [hunk 2]
```diff
@@ -733,6 +730,35 @@ private class PopupBlockingSettings: Setting {
     }
 }
 
+private class SaveLoginSettings: Setting {
```
Comment (fluffyemily) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: nit: there's a lot of duplicated code between all of the `UISwitch` based settings. Worth creating a subclass for SwitchSetting and then extending for `SaveLoginSettings`, `PopupBlockingSettings`, `ClosePrivateTabs`, `SendCrashReportsSetting` & `UseCompactTabLayoutSetting`?

Comment (fluffyemily) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: Also, maybe introduce some naming consistency here. Either all <name>Settings, all <name>Setting or all <name>.

I know this is cleaning up other peoples code, but it just niggles, you know?

Comment (jhugman) [inline Client/Frontend/Settings/SettingsTableViewController.swift]: Fair. Will happily clean up.

**Comment by jhugman** [comment]
@fluffyemily 03254b6 is reasonably big commit for just to be landed without further comment. Please could you give it a quick review before I squash and merge?

**Pull request merged**